p2p/proto/block.proto

```diff
@@ -0,0 +1,66 @@
+syntax = "proto3";
+import "p2p/proto/common.proto";
+import "google/protobuf/timestamp.proto";
+
+// for now, we assume a small consensus, so this fits in 1M. Else, these will be repeated
+message Signatures {
+    BlockID id = 1;
+    repeated Signature signatures = 2; // can be more explicit here about the signature structure as this is not part of account abstraction
+}
+
+// Note: commitments may change to be for the previous blocks like comet/tendermint
+// hash of block header sent to L1
+message BlockHeader {
+    BlockID parent_block = 1;
+
+    google.protobuf.Timestamp time = 2; // TODO: see if this needs to be Felt252 or can be converted
+
+    Address sequencer_address = 3;
+
+    Merkle state_diffs = 4; //  By order of (contract, key), taking last in case of duplicates.
+                            // This means the proposer needs to sort after finishing the block (TBD: patricia? )
+
+    // State is optional and appears every X blocks for the last block. This is to support
+    // snapshot sync and also so that light nodes can sync on state without state diffs.
+    Merkle state = 5; // hash of contract and class patricia tries. Same as in L1. Later more trees will be included
+
+    Hash proof_fact = 6; // for Kth block behind. A hash of the output of the proof
+
+    // The following merkles can be built on the fly while sequencing/validating txs.
+    Merkle transactions = 7; // By order of execution
+
+    Merkle events = 8; // By order of issuance.
+
+    Merkle receipts = 9; // By order of issuance.
```

Comment:
This is new, we need to define how it is calculated and if it affects the block hash.

Reply:
Yes, it affects the hash. WDYM by 'how it is calculated'?

Reply:
There is no defined way of hashing a receipt AFAIK. We need to define that first, right?

Reply:
I thought it is natural by the field order.

Reply:
Will define.